src/algorithms/trust_region.jl

```diff
@@ -24,9 +24,13 @@ function TrustRegion(; concrete_jac = nothing, linsolve = nothing, precs = DEFAU
         initial_trust_radius::Real = 0 // 1, step_threshold::Real = 1 // 10000,
         shrink_threshold::Real = 1 // 4, expand_threshold::Real = 3 // 4,
         shrink_factor::Real = 1 // 4, expand_factor::Real = 2 // 1,
-        max_shrink_times::Int = 32, vjp_autodiff = nothing, autodiff = nothing)
+        max_shrink_times::Int = 32, autodiff = nothing, vjp_autodiff = autodiff)
```

Comment:
This will cause problems if `autodiff = AutoForwardDiff()`. The nicer way would be to keep `vjp_autodiff = nothing` and check that if `autodiff` is FiniteDiff then set vjp_autodiff to that.

Reply:
makes sense. Updated.